### DISCUSSION TITLE
[Feature] Reduce Docker image size

### DISCUSSION THREAD

**Discussion by ngosang** [immich-app/immich#5604] [Feature Request]
### The feature

1. The `node_modules` folder included in the Docker image weights 200MB aprox. It can be reduced if you use some tool to bundle all JS dependencies together into a single JS. The same as web developers do.
* https://github.com/vercel/ncc
* https://github.com/ZenSoftware/bundled-nest#historical-reference

2. Pack the whole app as an executable. You don't need to install NodeJS/NPM in the Docker image. We should test this because the app start could be slower.
* https://github.com/vercel/pkg

Example:
```
npm i -g @vercel/ncc
npm install -g pkg
ncc build src/main.ts --out dist/ncc-main.js
pkg dist/ncc-main.js/index.js -o myapp -t latest-linux-x64
```

FYI @alextran1502 @martabal

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by martabal** [comment]
Interesting, I didn't know these tools. Thanks !

**Comment by jrasm91** [comment]
This most likely won't work since I believe the app still relies on other files in node_modules line binaries and such.
